### PR TITLE
[6.x] Add "missing" method to Request class

### PR DESCRIPTION
**Pull request by TitasGailius**
Just a simple method to check if a given key(s) is missing in the request object.

Instead of doing
```php
if (! $request->has('foo')) {
    //
}
```
you could do
```php
if ($request->missing('foo')) {
    //
}
```

### CODE REVIEW

File: src/Illuminate/Http/Concerns/InteractsWithInput.php [hunk 1]
```diff
@@ -91,6 +91,19 @@ public function has($key)
...
+     * Determine if the request contains a given input item key.
+     *
+     * @param  string|array  $key
+     * @return bool
+     */
+    public function missing($key)
+    {
+        $keys = is_array($key) ? $key : func_get_args();
+
+        return ! $this->has($keys);
```
Comment (xuanquynh) [inline src/Illuminate/Http/Concerns/InteractsWithInput.php]: What about this? Just forward arguments to has method.

```php
return ! $this->has(...func_get_args());
```

Comment (driesvints) [reply src/Illuminate/Http/Concerns/InteractsWithInput.php]: That won't work if the argument is an array.

**Comment by taylorotwell** [comment]
Might be worth PRing the laravel/docs repository to document this. Thanks.

File: src/Illuminate/Http/Concerns/InteractsWithInput.php (hunk 1 above)
Comment (xuanquynh) [reply src/Illuminate/Http/Concerns/InteractsWithInput.php]: @driesvints I am sure that it works properly. You can try this link https://repl.it/@QuynhNguyen/VapidImmaterialAccess

```php
<?php

class Request
{
  public function missing($key)
  {
    return $this->has(...func_get_args());
  }

  public function has($key)
  {
    $keys = is_array($key) ? $key : func_get_args();

    var_dump($keys);
  }
}

$request = new Request;

$request->missing('first'); // $keys = ['first']

$request->missing('first', 'second', 'third'); // $keys = ['first', 'second', 'third']

$request->missing(['first', 'second', 'third']); // $keys = ['first', 'second', 'third']
```

**Comment by Korko** [comment]
Shouldn't it be "misses" instead of "missing" to respect the same naming system as "has"?

**Comment by TitasGailius** [comment]
I think `missing` is fine. We can already do `cache()->missing()`.

**Comment by TheNewSound** [comment]
I find it ridiculous to make a separate method for 'missing()'. Totally unnecessary overhead. Increases laravel's size, documentation and project code becomes less 'unified' because of different syntax possibilities for exactly the same thing.

**Pull request merged**